r2dbc-spi/src/main/java/io/r2dbc/spi/Wrapped.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2017-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.r2dbc.spi;
+
+/**
+ * Indicates the class wraps(delegate, proxy, etc.) any of R2DBC SPI instances.
+ * <p>
+ * This interface provides a common way to retrieve the original object.
+ * Any wrapper(delegate, proxy, etc.) is required to implement this interface; so that, users
+ * can extract original object through this interface.
+ * <p>
+ * It is wrapper's responsibility to implement this.
+ *
+ * @author Tadaya Tsuyukubo
+ */
+public interface Wrapped {
+
+    /**
+     * Provides the original object.
+     *
+     * @return original object
+     */
+    Object unwrap();
```

Comment:
Wouldn't it be better to provide the type of what you're trying to unwrap and encapsulate that logic better?

Reply:
Oops, already [being discussed](https://github.com/r2dbc/r2dbc-spi/issues/28#issuecomment-443272315).

Reply:
Oh, I just made that comment almost same time :)

I might be over-thinking the usecase.
If you have opinion, please comment on #28.

Thanks,